### ISSUE TITLE
Connect sign up page to API

### ISSUE THREAD

**Issue by travissouthard** [Philadelphia-Lawyers-for-Social-Equity/PA_Expunger#23]
## Overview

Currently a sign up component exists but pushes data to localhost instead of actually sending it as a request to the API. Currently signup is effectively handled through the Django admin

### Requested Feature

Connect the sign up submit button to an API POST request to our API.
The existing endpoint `expunger/my-profile/` takes in the form data and on success returns the user data

**Comment by Michael-Dratch** [comment]
I had a few thoughts and questions about this issue, not sure these are questions that can be easily answered here or might require discussion.

The existing "/my-profile" endpoint provides the ability to  update a profile but not create a new user. I think a new endpoint should be added to create a user from scratch. 

I'd like to create a new endpoint "/signup" for this purpose

Additionally, it seems that creating a user profile requires an associated Attorney object and Organization object. Currently the sign up form does not have the required fields to create these additional objects. Is a profile required for all users? Can we just create a user on sign up and then redirect back to login? Should creating an attorney and selecting/creating an organization be part of the sign up flow? 

The sign up form also has a "role" drop down with the options "Lawyer", "Staff", and "Intern". Do all of these roles have an associated attorney?

**Comment by jgrimes86** [comment]
I agree that this issue requires more discussion, and I don't think it's worth working on this issue until there are decisions about (1) the questions you raise as well as my complaints in issue #46 and (2) how the app is going to be made available to PLSE (and potentially other non-profit legal aid organizations in the future)

My opinion on this issue is:  We should not have a signup page where users can independently create an account.  I know this is an open source project so anyone with enough know-how can clone the repo and start making their own petitions for expungement.  But when this app is deployed for use by PLSE, an admin should be in control of of creating authorized users.

Now for more (too much?) detail to justify my opinion and maybe answer your questions:

> it seems that creating a user profile requires an associated Attorney object and Organization object. Currently the sign up form does not have the required fields to create these additional objects. Is a profile required for all users? Can we just create a user on sign up and then redirect back to login? Should creating an attorney and selecting/creating an organization be part of the sign up flow?

Users don't need to have an ExpungerProfile when they sign up.  Their ExpungerProfile is created and an Attorney is associated with their ExpungerProfile when a user selects an attorney on the LandingPage component (`/` route).  PLSE is the default Organization automatically being associated with the ExpungerProfile when the profile is created.

Since the only Organization we're dealing with right now is PLSE, if the app is only deployed locally on PLSE computers, I guess it would be easy enough to have a new user create an account entirely through the signup page.  But what if the app is deployed as a website (which would make it much easier for PLSE staff/volunteers to use)?  And what happens if it is expanded so that other organizations can use it?  I don't think we want to let random people to sign up by selecting whatever organization they choose from a dropdown menu.  Allowing people to associate themselves with an organization without the organization's approval and to create petitions under that organization's header is a bad idea.

I think there needs to be some admin control over who has access, with admin-created users being able to edit their information later as needed.

>The existing "/my-profile" endpoint provides the ability to update a profile but not create a new user. I think a new endpoint should be added to create a user from scratch.
>
>I'd like to create a new endpoint "/signup" for this purpose

The `/my-profile` endpoint does have a POST method that seems to have been intended for creating new profiles, although the PUT method was used to do that instead.  I agree that a new endpoint to make a new user from scratch would make more sense.  But again, we need to be able to control what users are associated with an organization.  The only way I can think to do that is limiting users to those created through the admin page.

> The sign up form also has a "role" drop down with the options "Lawyer", "Staff", and "Intern". Do all of these roles have an associated attorney?

No.  An old migration creates "Attorney", "Intern", and "Volunteer" (not "Staff") groups, but they currently serve no real purpose because we aren't using the Django groups to manage permissions.  A user can be assigned to any group and still be associated with an Attorney entity.  Any user can also generate a petition for any attorney.  Selecting a different attorney on the `/` route just changes the user's ExpungerProfile to be linked to the new attorney.

For creating a user, the only real distinction is Attorney vs non-Attorney, because users who are attorneys in real life can be associated with an Attorney entity that stores their bar number.  But the "Attorney" user group has nothing to do with the Attorney model.

_If_ we're keeping the signup page and creating new users through it, it would be useful to have a user indicate whether they are an attorney and, if so, provide their bar number.  The backend route creating the user could also then create a new Attorney entity.  Without doing that automatically, the user/admin would still need to go to the Django admin page to manually create the Attorney entity.

**Comment by Michael-Dratch** [comment]
Thank you for all of the thorough response and context! I think it makes sense then to pause on this ticket.
